pypeit/find_objects.py

```diff
@@ -298,51 +334,63 @@ def run(self, std_trace=None, show_peaks=False):
-            # If they both shifted the same, there will be no reason to shift the tilts
-            tilt_flexure_shift = _spat_flexure - self.waveTilts.spat_flexure
-        else:
-            tilt_flexure_shift = self.spat_flexure_shift
-        msgs.info("Generating tilts image")
-        self.tilts = self.waveTilts.fit2tiltimg(self.slitmask, flexure=tilt_flexure_shift)
-
-        # Check if the user wants to use a pre-defined sky regions file.
-        skymask0, usersky = self.load_skyregions(None)
-        # Perform a first pass sky-subtraction without masking any objects. Should  we make this no_poly=True to
-        # have fewer degrees of freedom in the with with-object global sky fits??
-        initial_sky0 = self.global_skysub(skymask=skymask0, update_crmask=False, objs_not_masked=True).copy()
-        # First pass object finding
-        sobjs_obj, self.nobj = \
-            self.find_objects(self.sciImg.image-initial_sky0, self.sciImg.ivar, std_trace=std_trace,
-                              show_peaks=show_peaks,
-                              show=self.findobj_show and not self.std_redux,
-                              save_objfindQA=self.par['reduce']['findobj']['skip_second_find'] | self.std_redux)
-        # create skymask using first pass sobjs_obj
-        skymask_init = self.create_skymask(sobjs_obj)
-        # Check if the user wants to overwrite the skymask with a pre-defined sky regions file.
-        skymask_init, usersky = self.load_skyregions(skymask_init)
-
-        # If no objects were found and user did not define sky regions, don't redo global sky subtraction
-        if self.nobj == 0 and not usersky:
-            initial_sky = initial_sky0
-        else:
-            # Global sky subtract now using the skymask defined by object positions
-            initial_sky = self.global_skysub(skymask=skymask_init).copy()
-
-        # Second pass object finding on sky-subtracted image
-        if (not self.std_redux) and (not self.par['reduce']['findobj']['skip_second_find']):
-            sobjs_obj, self.nobj = self.find_objects(self.sciImg.image - initial_sky, self.sciImg.ivar,
+        # If the skip_skysub is set (i.e. image is already sky-subtracted), simply find objects
+        if self.par['reduce']['findobj']['skip_skysub']:
+            sobjs_obj, self.nobj = self.find_objects(self.sciImg.image, self.sciImg.ivar,
                                                      std_trace=std_trace, show=self.findobj_show,
                                                      show_peaks=show_peaks)
+            return np.zeros_like(self.sciImg.image), sobjs_obj
         else:
-            msgs.info("Skipping 2nd run of finding objects")
+            # Check if the user wants to use a pre-defined sky regions file.
+            user_regions = self.par['reduce']['skysub']['user_regions']
+            if user_regions == '': user_regions = None
```

Comment:
Does this change the value of `self.par['reduce']['skysub']['user_regions']`?

Reply:
@rcooke-ast can you please comment?